socorro/unittest/processor/test_mozilla_transform_rules.py

```diff
@@ -1619,38 +1739,26 @@ def test_rc(self):
         raw_dumps = {}
 
         with requests_mock.Mocker() as req_mock:
-            # Request for beta channel yields no data
-            req_mock.get(
-                self.API_URL + '?' + urlencode({
-                    'source.product': 'firefox',
-                    'build.id': '"20181018182531"',
-                    'target.channel': 'beta',
-                }),
-                json={
-                    'data': []
-                }
-            )
-
-            # Request for the same thing, but in the release channel yields
-            # release version number
-            req_mock.get(
-                self.API_URL + '?' + urlencode({
-                    'source.product': 'firefox',
-                    'build.id': '"20181018182531"',
-                    'target.channel': 'release',
-                    'like_target.version': '*rc*',
-                    '_limit': 1
-                }),
-                json={
-                    'data': [
-                        {
-                            'target': {
-                                'version': '63.0rc2'
-                            }
+            def payload_callback(request, context):
+                # The request for the beta channel returns no data
+                if 'beta' in request.text:
+                    return json.dumps({'hits': {'hits': []}})
+
+                # The request for the release channel returns a single item
+                if 'release' in request.text:
+                    return json.dumps({
+                        'hits': {
+                            'hits': [{
+                                '_source': {
+                                    'target': {
+                                        'version': '63.0rc2'
+                                    }
+                                }
+                            }]
                         }
-                    ]
-                }
-            )
+                    })
+
+            req_mock.post(self.API_URL, text=payload_callback)
```

Comment:
This one was tricky. We can't differentiate between the two posts enough to do `req_mock.post` for each one, so we have on mock call that has a payload callback that checks the request and returns the appropriate response. Then we later assert that the Buildhub query requests have the right shape and happen in the correct order.